### ISSUE TITLE
What does YOLOv8 change from 8.2 to 8.3.10 to 8.3.29?  

### ISSUE THREAD

**Issue by wjtan99** [ultralytics/ultralytics#17487]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

In a customized object detector training, I find that the model size changes dramatically from 8.2.48 to 8.3.10 to 8.3.29.    
There are 8 classes of objects in my training, I use a YOLOv8 small model (--pretrained yolov8s.pt) and the saved best.pt sizes and number of parameters on wandb:  

version | model file size | number of parameters 
8.2.48:  22 MB, 11.3M 
8.3.10: 19 MB, 9.84M 
8.3.29: 44 MB, 11.3M 

From simple calculation, 11.3M * 4 = 45.2 M.  So 8.3.29 uses FP32 but 8.2.48 and 8.3.10 use FP16. 

Furthermore, the number of parameters reduces from 11.3M in 8.2.48 to 9.84M in 8.3.10.  What are the key changes that make this big reduction in the number of parameters?  

Thanks a lot.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wjtan99, thank you for your interest in Ultralytics 🚀! We're excited to see your engagement with YOLOv8 and understand your curiosity about the model changes between these versions. 

For detailed changes across versions, you may want to look at the [Release Notes](https://github.com/ultralytics/ultralytics/releases) for YOLOv8, where each update is documented with technical modifications and improvements. This will give you specific insights into why you might be observing the differences in model sizes and parameters.

If this is a 🐛 Bug Report or you're encountering unexpected behavior, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it efficiently.

While this is an automated response, an Ultralytics engineer will review your question and assist you soon. In the meantime, we encourage you to join our community for more discussions:

- For real-time conversations, join our [Discord server](https://ultralytics.com/discord) 🎧.
- For in-depth technical discussions, use our [Discourse forum](https://community.ultralytics.com).
- You can also visit our [Subreddit](https://reddit.com/r/ultralytics) to share and gain knowledge with community members.

## Upgrade

To ensure you're using the latest features and improvements, consider upgrading to the latest `ultralytics` package. You can do this in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) using:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in various verified environments with all dependencies preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the status of our continuous integration tests, which ensure YOLO's functionality across different platforms. If it's green, all tests are passing successfully! 

Feel free to let us know if you have more questions or need further clarifications.

**Comment by Y-T-G** [comment]
8.2.48 and 8.3.29 are the same.

8.3.10 has a different head which was patched.

https://github.com/ultralytics/ultralytics/pull/17010

**Comment by wjtan99** [comment]
@Y-T-G Thanks for your quick response.  
However, the model size of the same training is doubled from 8.2.48 to 8.3.29. Is the default FP16 or FP32 mode changed from 8.2.48 to 8.3.29 in the object detector training?

**Comment by Y-T-G** [comment]
Did you complete the training? It should be the same at the end of the training.

**Comment by Y-T-G** [comment]
It's still the same size for me after the training with the latest version.

```python
In [5]: !ls -sh /ultralytics/runs/detect/train/weights/
total 13M
6.3M best.pt  6.3M last.pt

In [6]: !ls -sh yolov8n.pt
6.3M yolov8n.pt
```

**Comment by wjtan99** [comment]
@Y-T-G You are right. After the training is completed, the 8.3.2's model size went back to 22M, while it was 44M in the middle of training.  
Thank you again.
